runtime/devnet/src/config/api/mod.rs

```diff
@@ -163,53 +163,82 @@ impl<T: frame_system::Config> Contains<RuntimeRead> for Filter<T> {
 	use sp_core::crypto::AccountId32;
-	use RuntimeCall::Fungibles;
+	use RuntimeCall::{Balances, Fungibles};
+
+	use super::*;
 
 	const ACCOUNT: AccountId32 = AccountId32::new([0u8; 32]);
-	const CALLS: [RuntimeCall; 11] = [
-		Fungibles(transfer { token: 0, to: ACCOUNT, value: 0 }),
-		Fungibles(transfer_from { token: 0, from: ACCOUNT, to: ACCOUNT, value: 0 }),
-		Fungibles(approve { token: 0, spender: ACCOUNT, value: 0 }),
-		Fungibles(increase_allowance { token: 0, spender: ACCOUNT, value: 0 }),
-		Fungibles(decrease_allowance { token: 0, spender: ACCOUNT, value: 0 }),
-		Fungibles(create { id: 0, admin: ACCOUNT, min_balance: 0 }),
-		Fungibles(set_metadata { token: 0, name: vec![], symbol: vec![], decimals: 0 }),
-		Fungibles(start_destroy { token: 0 }),
-		Fungibles(clear_metadata { token: 0 }),
-		Fungibles(mint { token: 0, account: ACCOUNT, value: 0 }),
-		Fungibles(burn { token: 0, account: ACCOUNT, value: 0 }),
-	];
-
-	for call in CALLS {
-		assert!(Filter::<Runtime>::contains(&call))
+
+	#[test]
+	fn runtime_result_encode_works() {
+		let result = fungibles::ReadResult::<Runtime>::TotalSupply(1_000);
+		assert_eq!(RuntimeResult::Fungibles(result.clone()).encode(), result.encode());
+	}
+
+	#[test]
+	fn filter_prevents_runtime_filtered_calls() {
+		use pallet_balances::{AdjustmentDirection, Call::*};
+		use sp_runtime::MultiAddress;
+
+		const CALLS: [RuntimeCall; 4] = [
+			Balances(force_adjust_total_issuance {
+				direction: AdjustmentDirection::Increase,
+				delta: 0,
+			}),
+			Balances(force_set_balance { who: MultiAddress::Address32([0u8; 32]), new_free: 0 }),
+			Balances(force_transfer {
+				source: MultiAddress::Address32([0u8; 32]),
+				dest: MultiAddress::Address32([0u8; 32]),
+				value: 0,
+			}),
+			Balances(force_unreserve { who: MultiAddress::Address32([0u8; 32]), amount: 0 }),
+		];
+
+		for call in CALLS {
+			assert!(!Filter::<Runtime>::contains(&call))
+		}
+	}
+
+	#[test]
+	fn filter_allows_fungibles_calls() {
+		const CALLS: [RuntimeCall; 11] = [
+			Fungibles(transfer { token: 0, to: ACCOUNT, value: 0 }),
+			Fungibles(transfer_from { token: 0, from: ACCOUNT, to: ACCOUNT, value: 0 }),
+			Fungibles(approve { token: 0, spender: ACCOUNT, value: 0 }),
+			Fungibles(increase_allowance { token: 0, spender: ACCOUNT, value: 0 }),
+			Fungibles(decrease_allowance { token: 0, spender: ACCOUNT, value: 0 }),
+			Fungibles(create { id: 0, admin: ACCOUNT, min_balance: 0 }),
+			Fungibles(set_metadata { token: 0, name: vec![], symbol: vec![], decimals: 0 }),
+			Fungibles(start_destroy { token: 0 }),
+			Fungibles(clear_metadata { token: 0 }),
+			Fungibles(mint { token: 0, account: ACCOUNT, value: 0 }),
+			Fungibles(burn { token: 0, account: ACCOUNT, value: 0 }),
+		];
+
+		for call in CALLS {
+			assert!(Filter::<Runtime>::contains(&call))
+		}
+	}
+
+	#[test]
+	fn filter_allows_fungible_reads() {
+		use super::{fungibles::Read::*, RuntimeRead::*};
+		const READS: [RuntimeRead; 7] = [
```

Comment:
Unfortunately, this is not possible (or not easily possible). The reason is because it requires that underlying runtime types (such as `AccountId32`) also have to derive `enum_iterator::Sequence`.

Reply:
```
error[E0277]: the trait bound `sp_runtime::AccountId32: enum_iterator::Sequence` is not satisfied
  --> runtime/devnet/src/config/api/mod.rs:29:43
   |
29 | #[cfg_attr(test, derive(PartialEq, Clone, enum_iterator::Sequence))]
   |                                           ^^^^^^^^^^^^^^^^^^^^^^^ the trait `enum_iterator::Sequence` is not implemented for `sp_runtime::AccountId32`, which is required by `pallet_api::fungibles::Read<Runtime>: enum_iterator::Sequence`

```